### PR TITLE
Refactor network configuration for messages

### PR DESCRIPTION
**Pull request by maraoz**
This PR makes it easier to use bitcore-p2p with custom networks (such as testnet, or others)

### CODE REVIEW

File: lib/messages/message.js [hunk 1]
```diff
@@ -9,12 +11,12 @@ var Hash = bitcore.crypto.Hash;
...
  * @param {Object=} options
  * @param {String=} options.command
- * @param {Number=} options.magicNumber
+ * @param {Network=} options.network
  * @constructor
  */
 function Message(options) {
   this.command = options.command;
-  this.magicNumber = options.magicNumber;
+  this.network = options.network || Networks.defaultNetwork;
```
Comment (braydonf) [inline lib/messages/message.js]: The default should always be set in Messages factory, I don't think it needs to be set here if it's not defined.

**Comment by braydonf** [comment]
Tested. Should make things more clear.

File: docs/messages.md [hunk 2]
```diff
@@ -36,10 +36,10 @@ For advanced usage, you can also customize which constructor is used for Block a
 var messages = new Messages({Block: MyBlock, Transaction: MyTransaction});
 ```
 
-And additionally custom network magic:
+And additionally custom network:
```
Comment (braydonf) [inline docs/messages.md]: Existing typo: "And additionally a custom network:"

File: lib/messages/message.js [hunk 3]
```diff
@@ -22,7 +24,7 @@ function Message(options) {
  * @constructor
  */
 Message.prototype.toBuffer = Message.prototype.serialize = function() {
-
+  $.checkState(this.network, 'Need to have a defined network to serialize message');
```
Comment (braydonf) [inline lib/messages/message.js]: Should this check be on the constructor?

Comment (maraoz) [inline lib/messages/message.js]: No, some tests need to instantiate Message without parameters

**Pull request merged**